src/Evolution/Systems/GeneralizedHarmonic/BoundaryCorrections/UpwindPenalty.cpp

```diff
@@ -0,0 +1,292 @@
+  const size_t num_pts = char_speeds_int[0].size();
+  Variables<tmpl::list<::Tags::TempScalar<0>, ::Tags::TempScalar<1>,
+                       ::Tags::TempScalar<2>, ::Tags::TempScalar<3>,
+                       ::Tags::TempScalar<4>, ::Tags::TempScalar<5>,
+                       ::Tags::TempScalar<6>, ::Tags::TempScalar<7>>>
+      buffer(num_pts);
+  DataVector& weighted_lambda_spacetime_metric_int =
+      get(get<::Tags::TempScalar<0>>(buffer));
+  weighted_lambda_spacetime_metric_int = step_function(-char_speeds_int[0]);
+  DataVector& weighted_lambda_spacetime_metric_ext =
+      get(get<::Tags::TempScalar<1>>(buffer));
+  weighted_lambda_spacetime_metric_ext = -step_function(char_speeds_ext[0]);
+
+  DataVector& weighted_lambda_zero_int =
+      get(get<::Tags::TempScalar<2>>(buffer));
+  weighted_lambda_zero_int = step_function(-char_speeds_int[1]);
+  DataVector& weighted_lambda_zero_ext =
+      get(get<::Tags::TempScalar<3>>(buffer));
+  weighted_lambda_zero_ext = -step_function(char_speeds_ext[1]);
+
+  DataVector& weighted_lambda_plus_int =
+      get(get<::Tags::TempScalar<4>>(buffer));
+  weighted_lambda_plus_int = step_function(-char_speeds_int[2]);
+  DataVector& weighted_lambda_plus_ext =
+      get(get<::Tags::TempScalar<5>>(buffer));
+  weighted_lambda_plus_ext = -step_function(char_speeds_ext[2]);
+
+  DataVector& weighted_lambda_minus_int =
+      get(get<::Tags::TempScalar<6>>(buffer));
+  weighted_lambda_minus_int = step_function(-char_speeds_int[3]);
+  DataVector& weighted_lambda_minus_ext =
+      get(get<::Tags::TempScalar<7>>(buffer));
+  weighted_lambda_minus_ext = -step_function(char_speeds_ext[3]);
+
+  // D_spacetime_metric = Theta(-lambda_spacetime_metric^{ext})
+  // lambda_spacetime_metric^{ext} v_spacetime_metric^{ext}
+  //       - Theta(-lambda_spacetime_metric^{int}) lambda_spacetime_metric^{int}
+  //       v_spacetime_metric^{int}
+  // where the unit normals on both sides point in the same direction, out
+  // of the current element. Since lambda_spacetime_metric from the neighbor is
+  // computing with the normal vector pointing into the current element in the
+  // code, we need to swap the sign of lambda_spacetime_metric^{ext}. Theta is
+  // the heaviside step function with Theta(0) = 0.
+  for (size_t a = 0; a < Dim + 1; ++a) {
+    for (size_t b = a; b < Dim + 1; ++b) {
+      boundary_correction_spacetime_metric->get(a, b) =
+          weighted_lambda_spacetime_metric_ext *
+              char_speed_v_spacetime_metric_ext.get(a, b) -
+          weighted_lambda_spacetime_metric_int *
+              char_speed_v_spacetime_metric_int.get(a, b);
+
+      boundary_correction_pi->get(a, b) =
+          0.5 * (weighted_lambda_plus_ext * char_speed_v_plus_ext.get(a, b) +
+                 weighted_lambda_minus_ext * char_speed_v_minus_ext.get(a, b)) +
+          weighted_lambda_spacetime_metric_ext *
+              char_speed_constraint_gamma2_v_spacetime_metric_ext.get(a, b)
+
```

Comment:
I like the blank line before the `-` since I've found that at least I easily overlook a minus sign at the end of the line, and clang-format insists on the `-` being on its own line (which, I agree, is silly)